Objects/cown.c

```diff
@@ -0,0 +1,221 @@
+    atomic_store(&self->state, Cown_RELEASED);
+    self->owning_thread = 0;
+    sem_post(&self->semaphore);
+    Py_RETURN_NONE;
+}
+
+static PyObject *PyCown_get(PyCownObject *self, PyObject *args) {
+    bail_unless_acquired(self, "Cannot get value of unacquired cown");
+    if (self->value) {
+        Py_NewRef(self->value);
+        return self->value;
+    } else {
+        Py_RETURN_NONE;
+    }
+}
+
+// Needed to test for region object
+extern PyTypeObject PyRegion_Type;
+extern PyTypeObject PyCown_Type;
+
+static PyObject *PyCown_set(PyCownObject *self, PyObject *args) {
+    bail_unless_acquired(self, "Cannot set value of unacquired cown");
+
+    // Cowns are cells that hold a reference to a bridge object,
+    // (or another cown or immutable object)
+    if (args->ob_type == &PyRegion_Type ||
+        args->ob_type == &PyCown_Type ||
+        _Py_IsImmutable(args)) {
+        // Pyrona_ADDREFERENCE(_PyObject_CAST(self), args);
+
+        PyObject* old = self->value;
+        self->value = args;
+        Py_XINCREF(self->value);
+        Py_XDECREF(old);
+
+        return old ? old : Py_None;
+    } else {
+        // Invalid cown content
+        PyErr_SetString(PyExc_RuntimeError,
+            "Cowns can only store bridge objects, immutable objects or other cowns!");
+        return NULL;
+    }
+}
+
+static int PyCown_clear(PyCownObject *self) {
+    Py_CLEAR(self->value);
+    return 0;
+}
+
+static PyObject *PyCown_repr(PyCownObject *self) {
+#ifdef PYDEBUG
+    if (self->state == Cown_ACQUIRED) {
+        return PyUnicode_FromFormat(
+            "Cown(status=acquired by thread %zd,value=%S)",
+            PyThreadState_Get()->thread_id,
+            PyObject_Repr(self->value)
+        );
+    } else {
+        return PyUnicode_FromFormat(
+            "Cown(status=%s,value=%S)",
+            self->state == Cown_RELEASED
+                ? "released"
+                : "pending-release",
+            PyObject_Repr(self->value)
+        );
+    }
+#else
+    if (self->state == Cown_ACQUIRED) {
+        return PyUnicode_FromFormat(
+            "Cown(status=acquired by thread %zd)",
+            PyThreadState_Get()->thread_id
+        );
+    } else {
+        return PyUnicode_FromFormat(
+            "Cown(status=%s)",
+            self->state == Cown_RELEASED
+                ? "released"
+                : "pending-release"
+        );
+    }
+#endif
+}
+
+// Define the CownType with methods
+static PyMethodDef PyCown_methods[] = {
+    {"acquire", (PyCFunction)PyCown_acquire, METH_NOARGS, "Acquire the cown."},
+    {"release", (PyCFunction)PyCown_release, METH_NOARGS, "Release the cown."},
+    {"get",     (PyCFunction)PyCown_get,     METH_NOARGS, "Get contents of acquired cown."},
+    {"set",     (PyCFunction)PyCown_set,     METH_O, "Set contents of acquired cown."},
+    {NULL}  // Sentinel
+};
+
+
+PyTypeObject PyCown_Type = {
+    PyVarObject_HEAD_INIT(NULL, 0)
```

Comment:
Should this be a type?
```suggestion
    PyVarObject_HEAD_INIT(&PyType_Type, 0)
```